snapcraft/internal/project_loader.py

```diff
@@ -370,20 +370,23 @@ def _build_env_for_stage(stagedir, snap_name, confinement,
     return env
 
 
-def _get_snapcraft_yaml():
-    visible_yaml_exists = os.path.exists('snapcraft.yaml')
-    hidden_yaml_exists = os.path.exists('.snapcraft.yaml')
+def get_snapcraft_yaml():
+    possible_yamls = [
+        os.path.join('snap', 'snapcraft.yaml'),
+        'snapcraft.yaml',
+        '.snapcraft.yaml',
+    ]
 
-    if visible_yaml_exists and hidden_yaml_exists:
+    snapcraft_yamls = [y for y in possible_yamls if os.path.exists(y)]
+
+    if not snapcraft_yamls:
+        raise SnapcraftYamlFileError('snap/snapcraft.yaml')
+    elif len(snapcraft_yamls) > 1:
         raise EnvironmentError(
```

Comment:
This is actually an alias to `OSError`, which has an errno and stuff. We use it similarly in a few different places, so what do you think about making a `SnapcraftEnvironmentError` instead?